client/src/components/HomePage.jsx

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useState } from 'react';
+import '../styles/HomePage.css';
+import Post from './Post';
+const apiURL = import.meta.env.VITE_API_URL;
+
+const HomePage = () => {
+    const [allPosts, setAllPosts] = useState([])
+
+
+
+    useEffect(() => {
+        const fetchPosts = async () => {
+            const fetchAllPostsURL = `${apiURL}/api/posts`;
+            try {
+                const response = await fetch(fetchAllPostsURL);
+                if (!response.ok) {
+                    throw new Error("Failed to fetch data");
+                }
+                const body = await response.json();
+                setAllPosts(body);
+            } catch (error) {
+                console.error(error);
+            }
+        }
+        fetchPosts();
+    }, []);
```

Comment:
These useEffect calls, though used to call the fetch function initially, there's no dependency array passed to the useEffect, so it'll recreate the function at every render. We usually add a dependency, so that the useEffect is called whenever the dependency changes.

What you can do is to define the fetch calls in a useCallback, so the function is memoized. Then, call the callback function is called in the useEffect.
```suggestion
   const fetchPosts = useCallback(async () => {
   <your implementation>
    }, []);
    
    useEffect(() => {
        fetchPosts();
    }, [fetchPosts]);
```